library/datasort.c

```diff
@@ -560,6 +561,70 @@ static int datasort_split_iterator_free(struct eblob_iterate_control *ictl __att
 	return 0;
 }
 
+/*
+ * Create a view over a blob and insert it in dcfg->unsorted_list
+ * Assume that blob has already sorted
+ */
+static int datasort_add_view_chunk(struct datasort_cfg *dcfg, struct eblob_base_ctl * bctl) {
+	assert(dcfg);
+	assert(bctl);
+
+	int err = 0;
+
+	EBLOB_WARNX(dcfg->log, EBLOB_LOG_INFO, "defrag: datasort_add_view_chunk: start, name: %s",
+			bctl->name);
+
+	struct datasort_chunk *chunk = calloc(1, sizeof(*chunk));
+	if (chunk == NULL) {
+		EBLOB_WARNC(dcfg->log, EBLOB_LOG_ERROR, errno, "defrag: add_view_chunk: can't allocate memory for datasort_chunk");
+		return -errno;
+	}
+
+	eblob_base_wait_locked(bctl);
+	err = eblob_base_setup_data(bctl, 0);
+	if (err) {
+		goto err_out;
+	}
+
+	struct eblob_file_ctl *index_ctl = &bctl->index_ctl;
+	assert(index_ctl->fd > 0);
+
+	chunk->fd = bctl->data_ctl.fd;
+	chunk->already_sorted = 1;
+	chunk->base_view = 1;
+	chunk->offset = bctl->data_ctl.offset;
+	chunk->index_size = index_ctl->size;
+	chunk->count = chunk->index_size / sizeof(struct eblob_disk_control);
+	chunk->index = calloc(chunk->count, sizeof(struct eblob_disk_control));
+	if (chunk->index == NULL) {
+		EBLOB_WARNX(dcfg->log, EBLOB_LOG_ERROR, "defrag: load_index: can't allocate memory for index of size %zd",
+			chunk->count * sizeof(struct eblob_disk_control));
+		err = -errno;
+		goto err_out;
+	}
+
+	err = __eblob_read_ll(index_ctl->fd, chunk->index, index_ctl->size, 0);
+	if (err) {
+		EBLOB_WARNC(dcfg->log, EBLOB_LOG_ERROR, -err, "defrag: load_index: can't read index");
```

Comment:
```suggestion
		EBLOB_WARNC(dcfg->log, EBLOB_LOG_ERROR, -err, "defrag: datasort_add_view_chunk: can't read index");
```
or `__func__` could be used